plugins/cpp/service/src/diagram.cpp

```diff
@@ -329,14 +329,11 @@ std::string Diagram::getDetailedClassNodeLabel(const AstNodeInfo& nodeInfo_)
   _cppHandler.getReferences(nodes, nodeInfo_.id,
     CppServiceHandler::DATA_MEMBER, {});
 
-  std::map<core::AstNodeId, std::map<std::string, std::string>> props
-    = getProperties(nodes);
-
   for (auto it = nodes.begin(); it != nodes.end(); ++it)
   {
     std::string visibility = visibilityToHtml(*it);
     std::string content = memberContentToHtml(*it,
-      util::escapeHtml(it->astNodeValue + " : " + props[it->id]["Type"]));
+      util::escapeHtml(it->astNodeValue + " : " + getProperty(it->id, "Type")));
```

Comment:
Why did you change back the signature of the `getProperties` function? Now you do a lot of query which is not good for performance.